### PR TITLE
replace replay-verify with new db tool

### PR DESCRIPTION
**Pull request by areshand**
### Test Plan

python3 testsuite/replay_verify_run_local.py works

### CODE REVIEW

File: testsuite/replay_verify.py [hunk 1]
```diff
@@ -52,7 +52,8 @@ def replay_verify_partition(
     # run and print output
     process = subprocess.Popen(
         [
-            "target/release/replay-verify",
+            "target/release/aptos-db-tool",
```
Comment (rustielin) [inline testsuite/replay_verify.py:55]: while you're at it, can you change this line from `testsuite/replay-verify.py` to `testsuite/replay_verify.py` (- to _)? Replay verify is supposed to run on PRs that touch this file, but there's a typo that's preventing it from being triggered on PR:

https://github.com/aptos-labs/aptos-core/blob/main/.github/workflows/replay-verify.yaml#L24

**Review by rustielin — APPROVED** [review]
LGTM, stamping to unblock. Left a comment above about further testing

File: testsuite/replay_verify.py (hunk 1 above)
Comment (areshand) [reply testsuite/replay_verify.py:55]: updated the file name

Comment (rustielin) [reply testsuite/replay_verify.py:55]: Looks like that caught a bug: https://github.com/aptos-labs/aptos-core/actions/runs/4239339946/jobs/7367299441

**Review by rustielin — CHANGES_REQUESTED** [review]
See https://github.com/aptos-labs/aptos-core/pull/6721/files#r1113810922, looks like the CLI invocation is still not right

File: testsuite/replay_verify.py (hunk 1 above)
Comment (areshand) [reply testsuite/replay_verify.py:55]: add this new option that wasn't included in the db-tool previously

**Review by msmouse — APPROVED** [review]
test-replay canceled after 20mins run which is expected. Approving.

**Review by rustielin — APPROVED** [review]

File: testsuite/replay_verify.py (hunk 1 above)
Comment (msmouse) [reply testsuite/replay_verify.py:55]: > testsuite/replay-verify.py to testsuite/replay_verify.py

my bad and thank you

**Pull request merged**